server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteStoreMigrationSettingsUpdateIT.java

```diff
@@ -140,6 +143,37 @@ public void testNewRestoredIndexIsRemoteStoreBackedForRemoteStoreDirectionAndMix
         assertRemoteStoreBackedIndex(restoredIndexName2);
     }
 
+    // compatibility mode setting test
+
+    public void testSwitchToStrictMode() throws Exception {
```

Comment:
lets add a test for min max version check as well?

Reply:
Proceeding with UTs only for now, as discussed offline.